[R2] Added Postman guide (#4653)
cloudflare/cloudflare-docs: content/r2/examples/postman.md

## 3. Configure your R2 credentials

In the [Postman dashboard](https://www.postman.com/cloudflare-r2/workspace/cloudflare-r2/collection/20913290-14ddd8d8-3212-490d-8647-88c9dc557659?action=share&creator=20913290&ctx=documentation), select the **Cloudflare R2** collection and navigate to the **Variables** tab. In **Variables**, you can set variables within the R2 collection. They will be used to authenticate and interact with the R2 platform. Remember to always select **Save** after updating a variable.

![Select **Variables** in the Postman dashboard to continue with the example](/r2/static/postman-variables.png)

To execute basic operations, you must set the `account-id`, `r2-access-key`, and `r2-secret-id` variables in the Postman dashboard > **Variables**.

To do this:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?account=r2).
2. In **Account Home**, select **R2**.
3. In **R2**, under **Manage R2 API Tokens** on the right side of the dashboard, copy your Cloudflare account ID.

> 4. Go back to the Postman dashboard.

Reviewer: ```suggestion
4. Go back to the [Postman dashboard](https://www.postman.com/cloudflare-r2/workspace/cloudflare-r2/collection/20913290-14ddd8d8-3212-490d-8647-88c9dc557659?action=share&creator=20913290&ctx=documentation).
```

5. Set the **CURRENT VALUE** of `account-id` to your Cloudflare account ID and select **Save**.

Next, generate an R2 API token:

1. Go to the Cloudflare dashboard > **R2**.
2. On the right hand sidebar, select **Manage R2 API Tokens**.
3. Select **Create API token**.
4. Name your token **Postman** by selecting the pencil icon next to the API name and grant it the **Edit** permission.

![Generate an R2 API token for Postman by following the instructions above](/r2/static/postman-r2-api-token.png)

Guard this token and the **Access Key ID** and **Secret Access Key** closely. You will not be able to review these values again after finishing this step. Anyone with this information can fully interact with all of your buckets.

> After you have created you API token in the Cloudflare dashboard:

Reviewer: ```suggestion
After you have created your API token in the Cloudflare dashboard:
```

1. Go to the [Postman dashboard](https://www.postman.com/cloudflare-r2/workspace/cloudflare-r2/collection/20913290-14ddd8d8-3212-490d-8647-88c9dc557659?action=share&creator=20913290&ctx=documentation) > **Variables**.
2. Copy `Access Key ID` value from the Cloudflare dashboard and paste it into Postman’s `r2-secret-id` variable value and select **Save**.
3. Copy the `Secret Access Key` value value from the Cloudflare dashboard and paste into Postman’s `r2-access-key` variable value and select **Save**.

By now, you should have `account-id`, `r2-secret-id`, and `r2-access-key` set in Postman.

To verify the token:

1. In the Postman dashboard, select the **Cloudflare R2** folder dropdown arrow > **Buckets** folder dropdown arrow > **`GET`ListBuckets**.
2. Select **Send**.

The Postman collection uses AWS SigV4 authentication to complete the handshake.

> ![ListBuckets in Postman](/r2/static/postman-r2-list-buckets.png)

Reviewer: ```suggestion
![Follow the instructions above to access ListBuckets in the Postman dashboard](/r2/static/postman-r2-list-buckets.png)
```

---
title: Using R2 with Postman
summary: Learn how to configure Postman to interact with R2.
pcx-content-type: configuration
weight: 1001

> layout: example

Reviewer 2: @deadlypants1973 I noticed that this page is missing the TOC on the side, is that because of this field?

Reviewer 3: Yes :) All examples use this `layout`: https://github.com/cloudflare/cloudflare-docs/blob/production/layouts/_default/example.html Matches Workers' examples too

## 3. Configure your R2 credentials

Click on the `Cloudflare R2` collection and navigate to the `Variables` tab. Here, you can set variables within the R2 collection. They will be used to authenticate and interact with the R2 platform. Remember to always press Save after updating a variable.

![Postman Variables](/r2/static/postman-variables.png)

To execute some basic operations, we must set `account-id`, `r2-access-key`, and `r2-secret-id`.

Navigate to the [Cloudflare dashboard](https://dash.cloudflare.com/?account=r2) and select the account to use with R2. On the R2 overview page, copy your Cloudflare account ID in the from the right hand sidebar. Back in Postman, set the Current Value of `account-id` to your Cloudflare account ID and press Save.

Next, navigate back to the R2 overview page to set up your `r2-access-key` and `r2-secret-id`. On the right hand sidebar, click the _Manage R2 API Tokens_ link to create an R2 API Token called `Postman` and grant it the `Edit` permission.

![Generate an R2 API token for Postman](/r2/static/postman-r2-api-token.png)

> Guard this token closely. Anyone with this information can fully interact with all of your buckets.

Author: Yep, guarding of these values (but especially the secret) which comprise the token.

Once created, copy the token’s `Access Key ID` value into Postman’s `r2-secret-id` variable and the `Secret Access Key` value into Postman’s `r2-access-key` variable and press Save.

By now, you should have `account-id`, `r2-secret-id`, and `r2-access-key` set in Postman. To verify the token, navigate to the `ListBuckets` request and press `Send`. The Postman collection uses AWS SigV4 authentication to complete the handshake.

![ListBuckets in Postman](/r2/static/postman-r2-list-buckets.png)

You should see a `200 OK` response with a list of existing buckets. If you receive an error, ensure your R2 subscription is active and Postman variables are saved correctly.

## 4. Create a bucket

Now, let’s have some fun. In your `Variables` tab, think of a fun name for a bucket and set it as the `r2-bucket` variable and press Save. Then, select the `CreateBucket` request and press `Send`.

> ![CreateBucket in Postman](/r2/static/postman-r2-create-bucket.png)

Reviewer: ```suggestion
![Follow the instructions above to navigate to CreateBucket in the Postman dashboard](/r2/static/postman-r2-create-bucket.png)
```

You should see a `200 OK` response. If you run the `ListBuckets` request again, your bucket will appear in the list of results.

## 5. Add an object

A bucket isn’t very interesting without some stuff inside so let’s add an object. On the `Variables` tab, we set `r2-object` to `cat-pic.jpg` and press Save. Then, we select the `PutObject` request, and on the `Body` tab, we choose `binary` before attaching our favorite cat picture. Finally, we press `Send` to add a fluffy friend to R2.

> ![PutObject in Postman](/r2/static/postman-r2-put-object.png)

Reviewer: ```suggestion
![Follow the instructions above to PutObject in the Postman dashboard](/r2/static/postman-r2-put-object.png)
```

After a few seconds, you should receive a `200 OK` response.

## 6. Get an object

> It only takes one more click to download our cat friend using the `GetObject` request.

Author: Whoops, yep.  Looks like I ran out of steam here.

## 2. Explore R2 in Postman

Explore R2's publicly available [Postman collection](https://www.postman.com/cloudflare-r2/workspace/cloudflare-r2/collection/20913290-14ddd8d8-3212-490d-8647-88c9dc557659?action=share&creator=20913290). The collection is organized into a `Buckets` folder for bucket-level operations and an `Objects` folder for object-level operations. Operations in the `Objects > Upload` folder allow for adding new objects to R2.

## 3. Configure your R2 credentials

In the [Postman dashboard](https://www.postman.com/cloudflare-r2/workspace/cloudflare-r2/collection/20913290-14ddd8d8-3212-490d-8647-88c9dc557659?action=share&creator=20913290&ctx=documentation), select the **Cloudflare R2** collection and navigate to the **Variables** tab. In **Variables**, you can set variables within the R2 collection. They will be used to authenticate and interact with the R2 platform. Remember to always select **Save** after updating a variable.

![Select **Variables** in the Postman dashboard to continue with the example](/r2/static/postman-variables.png)

> To execute basic operations, you must set the `account-id`, `r2-access-key`, and `r2-secret-id` variables in the **Variables** tab.

Reviewer: ```suggestion
To execute basic operations, you must set the `account-id`, `r2-access-key`, and `r2-secret-id` variables in the Postman dashboard > **Variables**.
```

To do this:

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/?account=r2).
2. In **Account Home**, select **R2**.
3. In **R2**, under **Manage R2 API Tokens** on the right side of the dashboard, copy your Cloudflare account ID.
4. Go back to the Postman dashboard.
5. Set the **CURRENT VALUE** of `account-id` to your Cloudflare account ID and select **Save**.

Next, generate an R2 API token:

1. Go back to the Cloudflare dashboard > **R2**.
2. On the right hand sidebar, select **Manage R2 API Tokens**.
3. Select **Create API token**.

> 4. Name your token **Postman** by selecting the pencil icon next to the API name and grant it the `Edit` permission.

Reviewer: ```suggestion
4. Name your token **Postman** by selecting the pencil icon next to the API name and grant it the **Edit** permission.
```

![Generate an R2 API token for Postman by following the instructions above](/r2/static/postman-r2-api-token.png)

> Guard this token closely. Anyone with this information can fully interact with all of your buckets.

Reviewer: ```suggestion
Guard this token and the **Access Key ID** and **Secret Access Key** closely. You will not be able to review these values again after finishing this step. Anyone with this information can fully interact with all of your buckets.
```

3. Select the **Cloudflare R2** folder dropdown arrow > **Buckets** folder dropdown arrow > **`PUT`CreateBucket** and select **Send**.

![Follow the instructions above to navigate to CreateBucket in the Postman dashboard](/r2/static/postman-r2-create-bucket.png)

You should see a `200 OK` response. If you run the `ListBuckets` request again, your bucket will appear in the list of results.

## 5. Add an object

You will now add an object to your bucket:

1. Go to**Variables** in the Postman dashboard.
2. Set `r2-object` to `cat-pic.jpg` and select **Save**.
3. Select **Cloudflare R2** folder dropdown arrow > **Objects** folder dropdown arrow > **Multipart** folder dropdown arrow > **`PUT`PutObject** and select **Send**.
4. Go to **Body** and choose **binary** before attaching your cat picture.
5. Select **Send** to add the cat picture to your R2 bucket.

![Follow the instructions above to PutObject in the Postman dashboard](/r2/static/postman-r2-put-object.png)

After a few seconds, you should receive a `200 OK` response.

## 6. Get an object

It only takes a few more more clicks to download our cat friend using the `GetObject` request.

> 1. Select the **Cloudflare R2** folder dropdown arrow > **Objects** folder dropdown arrow > **`GET`GetObject** and select **Send**.

Reviewer: Do they need to do anything else on this screen other than pressing Send?

> 1. Select the **Cloudflare R2** folder dropdown arrow > **Objects** folder dropdown arrow > **`GET`GetObject** and select **Send**.

Author: Not unless we add "Enjoy the cute cat pic." 😆
